src/java.base/share/classes/java/util/random/RandomGenerator.java

```diff
@@ -137,6 +139,13 @@ static RandomGenerator of(String name) {
     static RandomGenerator getDefault() {
         return of("L32X64MixRandom");
     }
+    /**
+     * Returns a wrapper to use {@link java.util.Random}
+     * @return {@link java.util.Random}
```

Comment:
Possible reword:
"Returns a Random from this RandomGenerator."

It should be specified that the resulting Random acts in all respects except that
setSeed is not available.

Missing period at the end of the sentence.  Check all comments for grammar and punctuation.

Reply:
Is something like this better?
/**
     * Returns a {@link java.util.Random} from this {@link java.util.RandomGenerator}.
     * The resulting Random acts in all respects except that setSeed is not available.
     *
     * @return {@link java.util.Random}
     */

Reply:
Suggest:

> Returns an instance of {@link java.util.Random} based on this {@code RandomGenerator}.
> If this generator is already an instance of {@code Random}, it is returned. Otherwise, this method
> returns an instance of {@code Random} that delegates all methods except `setSeed` to this
> generator. Its `setSeed` method always throws {@link UnsupportedOperationException}.

(Note no link is necessary for RandomGenerator since we are in that class already.)

Reply:
Commited the change on df78e05e3e692e2189c9d318fbd4892a4b96a55f

Reply:
Will we gradually phase out the `Random` class? If yes, we should put this conversion method in `Random` so that we don't break the newer API shall `Random` be retired.

Reply:
I don't think Random will ever be removed, so this probably isn't an issue.

There is the overall question of which factoring is preferable: adding a default method on RandomGenerator, as is done here, or an alternative of adding a static utility to Random, something like:

    // Random.java
    public static Random from(RandomGenerator generator) { ... }

This makes the call site a bit longer and adds a level of nesting:

    Collections.shuffle(list, Random.from(RandomGenerator.of("L64X128MixRandom")));

compared to the default method approach:

    Collections.shuffle(list, RandomGenerator.of("L64X128MixRandom").asRandom()); 

I believe this allows the wrapper class to be placed in java.util, or even as a nested class of Random, instead of having to be a new top-level class in jdk.internal.

Another consideration is whether the adapter should be on the "new" thing or on the "old" thing. There's a little bit of precedent to put adapters on the "old" thing so that the "new" thing doesn't get polluted with the old. For example, consider Enumeration::asIterator. (Even though the adaptation is going the other way in that case, there are still alternative factorings that place the adapter on the new instead of the old.)

I don't think any of this is compelling but it's worth a discussion.